clang/include/clang/CIR/Dialect/IR/CIRTypes.td

```diff
@@ -155,21 +153,14 @@ def CIR_LongDouble : CIR_FloatType<"LongDouble", "long_double"> {
     format are all in use.
   }];
 
-  let parameters = (ins "mlir::Type":$underlying);
+  let parameters = (ins AnyTypeOf<[CIR_Double, CIR_FP80, CIR_FP128],
+    "expects !cir.double, !cir.fp80 or !cir.fp128">:$underlying);
```

Comment:
Can you add a test for this? It looks like the wording of the reported error will change, so we must not have a test now.